test/kube_checks.go

```diff
@@ -62,7 +64,7 @@ func WaitForDeploymentState(ctx context.Context, client kubernetes.Interface, na
 	})
 
 	if waitErr != nil {
-		return fmt.Errorf("deployment %q is not in desired state, got: %+v: %w", name, lastState, waitErr)
+		return errors.New(spew.Sprintf("deployment %q is not in desired state, got: %+v: %w", name, lastState, waitErr))
```

Comment:
Why move away from Errorf here? I'd have kept that and would propose:

```golang
return fmt.Errorf("deployment %q is not in desired state, error: %w, state:\n%s", name, waitErr spew.Sprint(lastState))

Reply:
It is another way to do it. It covers all other arguments too without thinking too much about them.

Reply:
I think I could have used `spew.Errorf()` to have only one call.